core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java

```diff
@@ -564,7 +564,7 @@ private JarBuildItem buildThinJar(CurateOutcomeBuildItem curateOutcomeBuildItem,
             Files.createDirectories(appDir);
             Files.createDirectories(quarkus);
             if (userProviders != null) {
-                Files.createDirectories(userProviders);
+                IoUtils.createOrEmptyDir(userProviders);
```

Comment:
This makes me uncomfortable given this directory could be configured as outside of the build tree.

I would rather only create the file below only if it's not already there (if that's the only problem you have).

Reply:
Thanks for the feedback, good to check only the file. I changed the code, the build is now running.

Reply:
Hmmm, but the whole point of my comment was to not change this call and not empty the directory :). If you empty the dir, the file will never be there for sure.